lib/puppet/provider/file_line/ruby.rb

```diff
@@ -43,7 +52,7 @@ def handle_create_with_match()
 
     File.open(resource[:path], 'w') do |fh|
       lines.each do |l|
-        fh.puts(regex.match(l) ? resource[:line] : l)
+        fh.puts(match_regex.match(l) ? resource[:line] : l)
```

Comment:
On second look, given this code path, you *should* be able to assume that `match_regex` is not `nil`.